substrate/frame/staking-async/src/pallet/impls.rs

```diff
@@ -934,6 +927,100 @@ impl<T: Config> Pallet<T> {
 	pub fn api_pending_rewards(era: EraIndex, account: T::AccountId) -> bool {
 		Eras::<T>::pending_rewards(era, &account)
 	}
+
+	/// Calculates the total stake of the lowest portion validators and stores it for the planned
+	/// era.
+	pub(crate) fn calculate_lowest_total_stake(era: EraIndex) {
+		// Only calculate if unbonding queue params have been set.
+		if let Some(params) = UnbondingQueueParams::<T>::get() {
+			// Determine the total stake from the lowest portion of validators and persist for the
+			// era.
+			let mut validator_total_stakes: Vec<_> =
+				ElectableStashes::<T>::get().into_iter().map(|(_, b)| b).collect();
+			let validators_to_check =
+				(params.lowest_ratio * validator_total_stakes.len() as u32).max(1) as usize;
+
+			// Sort exposure total stake by the lowest first and truncate to the lowest portion.
+			validator_total_stakes.sort();
+			validator_total_stakes.truncate(validators_to_check);
+
+			// Calculate the total stake of the lowest portion validators.
+			let total_stake = validator_total_stakes.into_iter().sum();
+
+			// Store the total stake of the lowest portion validators for the planned era.
+			Eras::<T>::set_lowest_stake(era, total_stake)
```

Comment:
Please confirm there are adequate tests (using realistic Polkadot/Kusama-scale validator sets) and try-state checks to ensure lowest_stake is calculated correctly.